tfjs-backend-webgl/src/webgl.ts

```diff
@@ -35,3 +36,136 @@ export {gpgpu_util, webgl_util};
 export function forceHalfFloat(): void {
   env().set('WEBGL_FORCE_F16_TEXTURES', true);
 }
+
+type TensorFromTextureConfig = {
+  texture: WebGLTexture,
+  shape: number[],
+  dtype: DataType,
+  texShapeRC: [number, number],
+  internalFormat: number,
+  textureFormat: number,
+  textureType: number
+};
+
+/**
+ * Create a TF.js tensor out of an existing WebGL texture. The texture is added
+ * to the WebGL texture registry.
+ *
+ * This makes it possible for TF.js applications to avoid GPU / CPU sync. For
+ * example, if your application includes a preprocessing step on the GPU, you
+ * could upload the GPU output directly to TF.js, rather than first downloading
+ * the values.
+ *
+ * ```js
+ * // Example for WebGL2:
+ * const gl = tf.backend().gpgpu.gl;
+ * const texture = gl.createTexture();
+ * const tex2d = gl.TEXTURE_2D;
+ * const width = 3;
+ * const height = 4;
+ *
+ * gl.bindTexture(tex2d, texture);
+ * gl.texParameteri(tex2d, gl.TEXTURE_WRAP_S, gl.CLAMP_TO_EDGE);
+ * gl.texParameteri(tex2d, gl.TEXTURE_WRAP_T, gl.CLAMP_TO_EDGE);
+ * gl.texParameteri(tex2d, gl.TEXTURE_MIN_FILTER, gl.NEAREST);
+ * gl.texParameteri(tex2d, gl.TEXTURE_MAG_FILTER, gl.NEAREST);
+ * gl.texImage2D(
+ *   tex2d, 0, gl.R32F, // internalFormat
+ *   width, height, 0,
+ *   gl.RED, // textureFormat
+ *   gl.FLOAT, // textureType
+ *   new Float32Array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]) // data
+ * );
+ *
+ * const logicalShape = [height, width];
+ * const physicalShape = [height, width];
+ * const a = tf.webgl.createTensorFromTexture(texture, logicalShape,
+ *   physicalShape);
+ *
+ * ```
+ *
+ * For postprocessing on the GPU, it's possible to retrieve the texture backing
+ * any tensor by calling the WebGL backend's `getTexture` method like so:
+ *
+ * ```js
+ * const a = tf.tensor1d([0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]);
+ * const tex = tf.backend().getTexture(a.dataId);
+ * ```
+ *
+ * @param obj An object with the following properties:
+ *  @param texture The WebGL texture to create a tensor from. The texture must
+ * be unpacked - each texel should only store a single value. The flattened
+ * values of the tensor will be read from left to right, and top to bottom. The
+ * texture can have empty texels at the end, but the values must be written
+ * densely - in other words, all empty texels must be contiguous.
+ *  @param shape The logical shape of the texture.
+ *  @param dtype The dtype of the tensor to be created.
+ *  @param texShapeRC The physical dimensions of the texture expressed as [rows,
```

Comment:
Is it better to use the same name with `gl.texImage2D`, like `width/height`? So you can say `The width of the texture provided to gl.texImage2D during texture creation.`
`The height of the texture provided to gl.texImage2D during texture creation.`. Similar to the other params (internalFormat -> internalformat, textureFormat -> format, textureType -> type)?

Reply:
done.